packages/eslint-plugin-formatjs/rules/no-explicit-icu-plural.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  RuleContext,
+  RuleListener,
+  RuleModule,
+} from '@typescript-eslint/utils/ts-eslint'
+import {TSESTree} from '@typescript-eslint/utils'
+import {extractMessages} from '../util'
+
+export const name = 'no-explicit-icu-plural'
+
+export type MessageIds = 'noExplicitIcuPlural'
+type Options = []
+
+// Matches `one {`, then zero or more `{1}` brace pairs, then a `1` before the closing brace `}`.
+const ruleRegex = /one\s*{[^{}]*(?:{[^}]*})*[^{}]*1/
```

Comment:
a possibly more robust implementation could use @formatjs/icu-message-parser - if you wanted to deal with an AST of the icu string instead of trying to regex search for patterns in it, that would be the way to do so.

Reply:
yeah agree. Other linter rules parses it, so you should be able to follow `enforce-plural-rule`